StyleCop.Analyzers/StyleCop.Analyzers.Test/SpacingRules/SA1012UnitTests.cs

```diff
@@ -0,0 +1,223 @@
+        public int TestProperty4{get; set; }
+    }
+}
+";
+
+            DiagnosticResult[] expected =
+            {
+                this.CSharpDiagnostic().WithLocation(6, 33).WithArguments(string.Empty, "preceded"),
+                this.CSharpDiagnostic().WithLocation(7, 34).WithArguments(string.Empty, "followed"),
+                this.CSharpDiagnostic().WithLocation(8, 33).WithArguments(string.Empty, "preceded"),
+                this.CSharpDiagnostic().WithLocation(8, 33).WithArguments(string.Empty, "followed"),
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will properly handle opening curly brackets in nested curly brackets.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestNestedCurlyBracketsAsync()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    using System.Collections.Generic;
+
+    public class TestClass
+    {
+        public void TestMethod()
+        {
+            new Dictionary<int, int> { { 1, 1 } };
+            new Dictionary<int, int>{ { 1, 1 } };
+            new Dictionary<int, int> {{ 1, 1 } };
+            new Dictionary<int, int> { {1, 1 } };
+            new Dictionary<int, int>{{1, 1 } };
+        }
+    }
+}
+";
+
+            DiagnosticResult[] expected =
+            {
+                this.CSharpDiagnostic().WithLocation(10, 37).WithArguments(string.Empty, "preceded"),
+                this.CSharpDiagnostic().WithLocation(11, 38).WithArguments(string.Empty, "followed"),
+                this.CSharpDiagnostic().WithLocation(11, 39).WithArguments(string.Empty, "preceded"),
+                this.CSharpDiagnostic().WithLocation(12, 40).WithArguments(string.Empty, "followed"),
+                this.CSharpDiagnostic().WithLocation(13, 37).WithArguments(string.Empty, "preceded"),
+                this.CSharpDiagnostic().WithLocation(13, 37).WithArguments(string.Empty, "followed"),
+                this.CSharpDiagnostic().WithLocation(13, 38).WithArguments(string.Empty, "preceded"),
+                this.CSharpDiagnostic().WithLocation(13, 38).WithArguments(string.Empty, "followed")
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that the analyzer will properly handle opening curly brackets in parentheses.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task TestParenthesisAsync()
+        {
+            var testCode = @"namespace TestNamespace
+{
+    public class TestClass
+    {
+        public void TestMethod()
+        {
+            TestMethod({ });
```

Comment:
:question: Is it even possible to have `({` in a valid source file?

:bulb: If you only need to test `})`, you can use `(new object[] { })`.